src/cljs/hello_world/game.cljs

```diff
@@ -91,6 +91,10 @@
           (js/setTimeout (:send-sprites-state-fn! websocket-message-send-functions) 300))
         this))))
 
+(defn- store-control-button-and-return-it [control-button]
+  (swap! util/game-state update :control-buttons conj control-button)
+  control-button)
```

Comment:
-  (swap! util/game-state update :control-buttons conj control-button)
+  (swap! util/game-state assoc :control-buttons control-button)
Could it be same as using assoc instead of update?? :)

Reply:
@flyjwayur it actually returns single button, `control-button` is the only argument, and it is added to the value of `:control-buttons` field of `game-state` atom.

Reply:
After that the original argument `control-button` is returned unchanged.

Reply:
@flyjwayur as you found out yourself, assoc and update is different in this case. we want to store all the buttons, so update is correct here, if we use assoc here, previous button is replaced by the next one and in the end only the last button will be stored.